### DISCUSSION TITLE
Welcome to StoManager1 Discussions!

### DISCUSSION THREAD

**Discussion by JiaxinWang123** [JiaxinWang123/StoManager1#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by zhiyuanliu-cau** [comment]
Hi, it gave me error a file named "best.pt" could not be found and no stoma detected. May I know why? I have attached error info and my stoma picture. May I get your help?
![error](https://github.com/user-attachments/assets/fd2b345c-9c5c-4974-8203-5d53a1892859)
![WT-2-400X](https://github.com/user-attachments/assets/de6d6922-e333-4a07-b343-83f4c7010289)
![WT-3-400X](https://github.com/user-attachments/assets/0b5bcdb4-04b1-4bff-9730-208771fda619)

**Comment by JiaxinWang123** [comment]
Hi,

The first error you encountered is because the best.bt file was deleted, if you trained your own model, you should also name it as best.bt. If you couldn't find it, please download it from here<https://mstate-my.sharepoint.com/:f:/g/personal/jw3994_msstate_edu/EjzgRGSJ83ZOuUyy1tG1NtgBcwofVOIXvnvTTvWaPD3VCQ?e=Bokpao>. Or you can redownload the package. If the current model does not work well for your case, you can train your own model by labeling your image in Roboflow and train your model in StoManager1 following the manual.

Good luck,

Jiaxin


Jiaxin Wang, Ph.D.

Department of Forestry

Mississippi State University

Thompson Hall 350C

(662)-518-3881

Personal email: ***@***.******@***.***>

Homepage<https://www.jiaxin-wang.com/>



________________________________

**Comment by JiaxinWang123** [comment]
The file should be best.pt. Sorry about the confusion.

Thanks,
Jiaxin


Jiaxin Wang, Ph.D.

Department of Forestry

Mississippi State University

Thompson Hall 350C

(662)-518-3881

Personal email: ***@***.******@***.***>

Homepage<https://www.jiaxin-wang.com/>



________________________________

**Reply by zhiyuanliu-cau** [reply]
Thanks for your help! Let me try it! Appreciated!

**Comment by JiaxinWang123** [comment]
Hi,

If you are using StoManager1 to detect stomata, you need to download and run StoManager1_v10_new.rar<https://zenodo.org/records/10553682/files/StoManager1_v10_new.rar?download=1>  (GPU version) or StoManager1_v10_new_CPU.rar<https://zenodo.org/records/10553682/files/StoManager1_v10_new_CPU.rar?download=1>  (CPU version). If you need to train model using your own dataset, you will also need to download model_training_in_app.rar<https://zenodo.org/records/10553682/files/model_training_in_app.rar?download=1>. The model_training_in_app is not a standalone app, it can only be used within StoManager1.
If you could not download the app from zenodo, you can download them from figshare. Here are the link: https://figshare.com/articles/software/StoManager1_rar/22205020

Best,
Jiaxin


Jiaxin Wang, Ph.D.

Department of Forestry

Mississippi State University

Thompson Hall 350C

(662)-518-3881

Personal email: ***@***.******@***.***>

Homepage<https://www.jiaxin-wang.com/>



________________________________
From: ZhouYongxin-12345 ***@***.***>
Sent: Monday, June 9, 2025 8:00 PM
To: JiaxinWang123/StoManager1 ***@***.***>
Cc: Wang, Jiaxin ***@***.***>; Author ***@***.***>
Subject: [BULK] Re: [JiaxinWang123/StoManager1] Welcome to StoManager1 Discussions! (Discussion #1)


Subject: Inquiry Regarding an Error in StoManager1 v1.0.0
Dear Professor Wang,
I hope this email finds you well. My apologies for reaching out unexpectedly.
I am a student currently using StoManager1 v1.0.0 for stomatal measurement in plant leaves. However, I encountered an error while running the software and would appreciate your guidance in resolving it.
Error Details：When launching the software, the following error occurs:
text
Traceback (most recent call last):
File "model_training_in_app.py", line 18, in
IndexError: list index out of range
Operating System: Windows 11
Attempted Solutions: Reinstalled the software
Request for Assistance
Could you kindly advise on the possible causes of this error? If additional logs or information are needed, please let me know—I’d be happy to provide further details.
Thank you for your time and support. I look forward to your reply.
Best regards,
Personal email: ***@***.******@***.***>
1.png (view on web)<https://github.com/user-attachments/assets/d071edfa-93cf-4527-aee9-a9546d073a50>
2.png (view on web)<https://github.com/user-attachments/assets/79735ab1-cf1c-4efb-90fd-9653eee26c09>
